pkg/controller/perconaservermongodb/users.go

```diff
@@ -0,0 +1,241 @@
+	if cr.Status.State != api.AppStateReady {
+		return nil
+	}
+
+	newSysData, err := json.Marshal(sysUsersSecretObj.Data)
+	if err != nil {
+		return errors.Wrap(err, "marshal sys secret data")
+	}
+	newSecretDataHash := sha256Hash(newSysData)
+	dataChanged, err := sysUsersSecretDataChanged(newSecretDataHash, &internalSysSecretObj)
+	if err != nil {
+		return errors.Wrap(err, "check sys users data changes")
+	}
+
+	if !dataChanged {
+		return nil
+	}
+
+	restartSfs, err := r.manageSysUsers(cr, &sysUsersSecretObj, &internalSysSecretObj)
+	if err != nil {
+		return errors.Wrap(err, "manage sys users")
+	}
+
+	internalSysSecretObj.Data = sysUsersSecretObj.Data
+	err = r.client.Update(context.TODO(), &internalSysSecretObj)
+	if err != nil {
+		return errors.Wrap(err, "update internal sys users secret")
+	}
+
+	if restartSfs {
+		sfsTemplateAnnotations["last-applied-secret"] = newSecretDataHash
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaServerMongoDB) manageSysUsers(cr *api.PerconaServerMongoDB, sysUsersSecretObj, internalSysSecretObj *corev1.Secret) (bool, error) {
+	var (
+		sysUsers  []sysUser
+		userAdmin *sysUser
+	)
+	restartSfs := false
+	userAdminUserExist := false
+	userAdminPasswordExist := false
+	for key := range sysUsersSecretObj.Data {
+		switch key {
+		case envMongoDBUserAdminPassword:
+			userAdminPasswordExist = true
+		case envMongoDBUserAdminUser:
+			userAdminUserExist = true
+		}
+
+		if bytes.Compare(sysUsersSecretObj.Data[key], internalSysSecretObj.Data[key]) == 0 {
+			continue
+		}
+
+		switch key {
+		case envMongoDBBackupPassword:
+			sysUsers = append(sysUsers, sysUser{
+				Name: string(sysUsersSecretObj.Data[envMongoDBBackupUser]),
+				Pass: string(sysUsersSecretObj.Data[envMongoDBBackupPassword]),
+			},
+			)
+			restartSfs = true
+		case envMongoDBClusterAdminPassword:
+			sysUsers = append(sysUsers, sysUser{
+				Name: string(sysUsersSecretObj.Data[envMongoDBClusterAdminUser]),
+				Pass: string(sysUsersSecretObj.Data[envMongoDBClusterAdminPassword]),
+			},
+			)
+		case envMongoDBClusterMonitorPassword:
+			sysUsers = append(sysUsers, sysUser{
+				Name: string(sysUsersSecretObj.Data[envMongoDBClusterMonitorUser]),
+				Pass: string(sysUsersSecretObj.Data[envMongoDBClusterMonitorPassword]),
+			},
+			)
+			if cr.Spec.PMM.Enabled {
+				restartSfs = true
+			}
+		case envMongoDBUserAdminPassword:
+			userAdmin = &sysUser{
+				Name: string(sysUsersSecretObj.Data[envMongoDBUserAdminUser]),
+				Pass: string(sysUsersSecretObj.Data[envMongoDBUserAdminPassword]),
+			}
+		case envPMMServerPassword:
+			restartSfs = true
+		}
+	}
+	if !userAdminUserExist || !userAdminPasswordExist {
+		return restartSfs, errors.New("userAdminUser or userAdminPassword not exist in secret " + sysUsersSecretObj.Name)
+	}
+	if userAdmin != nil {
+		sysUsers = append(sysUsers, *userAdmin) // UserAdmin must be the last in array because we use him for mongo client connection
```

Comment:
```suggestion
		sysUsers = append(sysUsers, *userAdmin) // userAdmin must be the last user to update since we're using it for the mongo connection
```